packages/store/src/StoreCore.sol

```diff
@@ -811,14 +811,16 @@ library StoreCoreInternal {
     bytes32[] memory keyTuple,
     uint8 schemaIndex
   ) internal pure returns (uint256) {
-    return uint256(keccak256(abi.encode(SLOT, tableId, keyTuple, schemaIndex)));
+    // offset by 1 to avoid collision with static data
+    // assumes schemaIndex < 255
+    return uint256(SLOT ^ tableId ^ bytes1(schemaIndex + 1) ^ keccak256(abi.encodePacked(keyTuple)));
   }
 
   /**
    * Compute the storage location for the length of the dynamic data
    */
   function _getDynamicDataLengthLocation(bytes32 tableId, bytes32[] memory keyTuple) internal pure returns (uint256) {
-    return uint256(keccak256(abi.encode(SLOT, tableId, keyTuple, "length")));
+    return uint256(SLOT ^ tableId ^ bytes32("length") ^ keccak256(abi.encodePacked(keyTuple)));
```

Comment:
what if i register a table with id `tableIdA` which equals `tableIdB ^ bytes32(length)`? would that create a conflict between `_getStaticDataLocation` for `tableA` and `_getDynamicDataLengthLocation` for `tableB`?

I wonder if, since `tableId` is also a user influenceable parameter we need to also hash `tableId`?

Reply:
Ah ya forgot that tableId is a string instead of a hash... that's a problem.

Reply:
Should create a hash for "length" like SLOT.